parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java

```diff
@@ -104,11 +104,12 @@ public void testEmptyArray() throws Exception {
       writer.write(record);
     }
 
-    AvroParquetReader<GenericRecord> reader = new AvroParquetReader<GenericRecord>(testConf, file);
-    GenericRecord nextRecord = reader.read();
+    try (AvroParquetReader<GenericRecord> reader = new AvroParquetReader<>(testConf, file)) {
```

Comment:
I think you are fine-tuning how to put space after try for each source code. 
But isn't this different only here?

Reply:
I don't think there is a clear consensus on this:
```
MacBook-Pro-van-Fokko:parquet-mr fokkodriesprong$ grep -R "try (" .
./parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java:    try (AvroParquetReader<GenericRecord> reader = new AvroParquetReader<>(testConf, file)) {
./parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java:    try (ParquetWriter<GenericData.Record> writer = AvroParquetWriter
./parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java:    try (ParquetReader<GenericRecord> reader = AvroParquetReader
./parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWriteOldListBehavior.java:    try (AvroParquetWriter<GenericRecord> writer = new AvroParquetWriter<>(file, schema)) {
./parquet-avro/src/test/java/org/apache/parquet/avro/TestArrayCompatibility.java:    try (ParquetFileReader reader = ParquetFileReader.open(new Configuration(), test)) {
./parquet-avro/src/test/java/org/apache/parquet/avro/AvroTestUtil.java:    try (ParquetReader<D> fileReader = AvroParquetReader
./parquet-avro/src/test/java/org/apache/parquet/avro/AvroTestUtil.java:    try (ParquetWriter<D> writer = AvroParquetWriter
./parquet-tools/src/test/java/org/apache/parquet/tools/read/TestSimpleRecordConverter.java:    try (
./parquet-tools/src/test/java/org/apache/parquet/tools/read/TestSimpleRecordConverter.java:    try (
./parquet-tools/src/main/java/org/apache/parquet/tools/command/ColumnIndexCommand.java:    try (ParquetFileReader reader = ParquetFileReader.open(in)) {
./parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java:    try (SeekableInput in = openSeekable(source)) {
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/ToAvroCommand.java:      try (DataFileWriter<Record> writer = w.create(projection, create(outputPath))) {
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/ConvertCommand.java:      try (ParquetWriter<Record> writer = AvroParquetWriter
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/ConvertCSVCommand.java:    try (AvroCSVReader<Record> reader = new AvroCSVReader<>(
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/ConvertCSVCommand.java:      try (ParquetWriter<Record> writer = AvroParquetWriter
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/SchemaCommand.java:        try (OutputStream out = create(outputPath)) {
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/SchemaCommand.java:    try (SeekableInput in = openSeekable(source)) {
./parquet-cli/src/main/java/org/apache/parquet/cli/commands/ShowColumnIndexCommand.java:    try (ParquetFileReader reader = ParquetFileReader.open(in)) {
./parquet-hadoop/src/test/java/org/apache/parquet/filter2/recordlevel/PhoneBookWriter.java:    try (ParquetWriter<Group> writer = builder.build()) {
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestMultipleWriteRead.java:    try (ParquetWriter<Group> writer = ExampleParquetWriter.builder(file)
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestMultipleWriteRead.java:    try (ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), file).build()) {
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestMultipleWriteRead.java:    try (ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), file)
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetFileWriter.java:    try (ParquetFileReader reader = new ParquetFileReader(HadoopInputFile.fromPath(path, configuration),
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetWriter.java:    try (ParquetWriter<Group> writer = ExampleParquetWriter.builder(path)
./parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetWriter.java:    try (ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), path).build()) {
./parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java:    try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
./parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java:    try (SeekableInputStream in = file.newStream()) {
./parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java:    try (ParquetFileReader reader = ParquetFileReader.open(file)) {
./parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ColumnIndexFilterUtils.java:      try (Formatter formatter = new Formatter()) {
./parquet-benchmarks/src/main/java/org/apache/parquet/benchmarks/FilteringBenchmarks.java:      try (ParquetWriter<Group> writer = writeConfigurator.configureBuilder(ExampleParquetWriter.builder(file)
./parquet-benchmarks/src/main/java/org/apache/parquet/benchmarks/FilteringBenchmarks.java:    try (ParquetReader<Group> reader = context.createReaderBuilder()
./parquet-benchmarks/src/main/java/org/apache/parquet/benchmarks/NestedNullWritingBenchmarks.java:    try (ParquetWriter<Group> writer = ExampleParquetWriter.builder(BLACK_HOLE)
./parquet-column/src/main/java/org/apache/parquet/internal/column/columnindex/ColumnIndexBuilder.java:      try (Formatter formatter = new Formatter()) {
./parquet-column/src/main/java/org/apache/parquet/internal/column/columnindex/OffsetIndexBuilder.java:      try (Formatter formatter = new Formatter()) {
./parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/ByteBasedBitPackingGenerator.java:    try (FileWriter fw = new FileWriter(file)) {
./parquet-generator/src/main/java/org/apache/parquet/encoding/bitpacking/IntBasedBitPackingGenerator.java:    try (FileWriter fw = new FileWriter(file)) {
```

From a SE point of view, I would say that either of them is fine since there is no enforcing on the rule. However, I think without space is neater. WDYT?

Reply:
I also think that there is no policy for the whole source code about this style. I thought you were adjusting for each source code style. This issue is withdrawn.